### PR TITLE
DaemonState: `write_on_change`

### PR DESCRIPTION
**Pull request by Buckram123**
This PR aims to add write_on_change flag to DaemonState with default to `true` for auto-updating state, instead of waiting for program to finish execution
### Checklist

- [x] Changelog updated.
- [ ] Docs updated.

### CODE REVIEW

**Review by Kayanski — APPROVED** [review]
Otherwise very nice !

File: packages/clone-testing/src/state.rs [hunk 1]
```diff
@@ -29,6 +29,7 @@ impl MockState {
                 chain,
                 deployment_id.to_string(),
                 true,
+                true,
```
[the remark this answers is not in the document]
Comment (Buckram123) [reply packages/clone-testing/src/state.rs]: Yeah, read-only + write_on_save is funny combination

[the remark this answers is not in the document]
Comment (Buckram123) [reply packages/clone-testing/src/state.rs]: Fixed!

**Review by CyberHoward — APPROVED** [review]

File: packages/clone-testing/src/state.rs (hunk 1 above)
[the remark this answers is not in the document]
Comment (Buckram123) [reply packages/clone-testing/src/state.rs]: @Kayanski https://github.com/AbstractSDK/abstract/actions/runs/9515882159/job/26231034923?pr=376#step:8:1136 integration tests failing, clone-testing needs new release as well

I can do it, it should be versioned as 0.5.2, right?

**Comment by Kayanski** [comment]
Sure, go ahead. Check crates.io for the current version

**Pull request merged**